# Notebook Instructions

1. All the <u>code and data files</u> used in this course are available in the downloadable unit of the <u>last section of this course</u>.
2. You can run the notebook document sequentially (one cell at a time) by pressing **shift + enter**. 
3. While a cell is running, a [*] is shown on the left. After the cell is run, the output will appear on the next line.

This course is based on specific versions of python packages. You can find the details of the packages in <a href='https://quantra.quantinsti.com/quantra-notebook' target="_blank" >this manual</a>.

## Expressions

'Expressions' are generally a combination of numbers, variables, and operators. <br>
<br>
In this IPython notebook, we will make use of Expressions to understand the TVM concepts. 

### Future value (FV)

What would be the FV, if I have $1000 with me now and I will be investing it for 1 year, at an annual return of 5%? 

In [1]:
PV = 1000
r = 0.05
n = 1

# Formula for calculating Future Value
FV = PV * ((1+r) ** n)

print(FV)

1050.0


### Present value (PV)

What would be the PV, if I have to discount $1050 at a 5% annual rate for a period of 1 year?

In [2]:
FV = 1050
r = 0.05
n = 1

# Formula for calculating Present Value
PV = FV / ((1 + r) ** n)

print(PV)

1000.0


### Compounding 

Assume that the 5% annual interest rate bond makes semiannual payments. That is, for an investment of $1000, you will get 25 dollars, after the first 6 months and another 25 dollars after 1 year. The annual rate of interest is 5%. What would be the FV, if I hold the bond for 1 year? 

In [3]:
PV = 1000
r = 0.05
n = 2  # number of periods = 2 since bond makes semiannual payments
t = 1  # number of years

# Formula for compounding
FV = PV * ((1+(r/n)) ** (n*t))

print(FV)

1050.625


### Annuity payments

What would be the annual periodic saving amount, if you want a lump sum of $9476.96 at the end of 3 years? The rate of return is 10%? (Assuming the first payment is made at the start of each year).<br>
<br>
(This is one of the required calculation from 'PDF: TVM Applications' unit) 

In [4]:
r = 0.1
n = 3
PV = 0
FV = 9476.96

# Formula for Annuity payments, given Future Value
AP = (FV * r) / (((1 + r) ** n - 1)*(1+r))
print(AP)

2602.8453721505043


What would be the PV, given a cash outflow of $2500 for a period of 5 years and rate of return being 10%?<br>
<br>
(This is one of the required calculation from 'PDF: TVM Applications' unit) 

In [5]:
r = 0.1
n = 5
AP = 2500

# Formula for PV, given Annuity payments
PV = (AP * (1 - ((1 + r) ** -n))) / r
print(PV)

9476.966923521126


What would be the PV, given a cash outflow of $30,000 for a period of 45 years and rate of return being 8%?<br>
<br>
(This is one of the required calculation from 'PDF: TVM Applications' unit) 

In [6]:
r = 0.08
n = 45
AP1 = 30000

# Formula for PV, given Annuity payments
PV = (AP1 * (1 - ((1 + r) ** -n))) / r
print(PV)

363252.0450945144


What would be the annual saving amount (AP), if you want to save a lump sum of $363252.045 in 25 years and rate of return being 15%?<br>
<br>
(This is one of the required calculation from 'PDF: TVM Applications' unit) 

In [7]:
r = 0.15
n = 25
PV = 0
FV = 363252.045095

# Formula to calculate Annuity Payments, given FV
AP = (FV * r) / (((1 + r) ** n - 1)*(1+r))

# Formula to calculate Annuity Payments, given PV
#AP = (r * PV) / (1 - ((1 + r) ** -n))

print(AP)

1484.4065248166885


These are some ways, one can use Expressions. <br>

In the upcoming section, we will learn about Python <b>Data Structures</b>.<br><br>